### DISCUSSION TITLE
ConnectionLinetype changes to bezier after conneting nodes

### DISCUSSION THREAD

**Discussion by badiuciprian** [xyflow/xyflow#1398] [Q&A]
HI, first of all sorry if there is already a discussion about this but I could not find it.
I'm new to react and I tried to create a demo based on the examples you provided.
The issue is that I've set the connectionLineType={ConnectionLineType.Step} in the ReactFlow properties and it shows the line as step while connecting 2 nodes, but as soon as the connection is made, it switches to bezier. 
Am I doing something wrong?
Thanks a lot.

**Comment by moklick** [comment]
You need to adjust the `onConnect` handler in order to pass the correct type:

```js
const onConnect = (params) => setElements((els) => addEdge({ ...params, type: 'step' }, els));
```

**Reply by moklick** [reply]
Or you can overwrite the edgeTypes:

```js 
import { StepEdge } from 'react-flow-renderer';

...

const edgeTypes = {
  default: StepEdge
};

...

<ReactFlow edgeTypes={edgeTypes} /> 
```

**Reply by badiuciprian** [reply]
Thanks a lot, it worked as expected,
(I new I did something wrong, I feel so dumb now :) )

**Reply by moklick** [reply]
no need to feel dumb here.. just blame the docs :D

**Reply by badiuciprian** [reply]
:))
